### ISSUE TITLE
ExceptionError when dropping column in SQLite

### ISSUE THREAD

**Issue by troccoli** [doctrine/dbal#5606]
### Bug Report

|    Q        |   A
|------------ | ------
| Version     | 3.4.1

#### Summary

ErrorException thrown when dropping a column in SQLite

#### Current behaviour

I'm currently using the package in a Laravel project. For my test I use a SQLite database. After upgrading from 3.3.8 my tests started failing with the following error

```
ErrorException: Undefined array key -1

/home/runner/work/tubecrush/tubecrush/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php:257
/home/runner/work/tubecrush/tubecrush/vendor/doctrine/dbal/src/Schema/SqliteSchemaManager.php:605
```

The issue is acutally in `\Doctrine\DBAL\Schema\SqliteSchemaManager::getForeignKeyDetails()` which returns an empty array when it shouldn't.

#### How to reproduce

As I said I narrowed it down to the `getForeignKeyDetails` method.

Through some debugging I have found out that `getCreateTableSQL()` returns ```CREATE TABLE "posts" ("id" integer not null primary key autoincrement, "title" varchar not null, "content" text not null, "author_id" integer not null, "deleted_at" datetime, "created_at" datetime, "updated_at" datetime, "photo" varchar not null default 'photos/dummy.png', "line_id" integer not null default '0', "photo_credit" varchar, "likes" integer not null default '0', "slug" varchar, "status" varchar check ("status" in ('draft', 'published')) not null default 'draft', "published_at" datetime, foreign key("author_id") references "users"("id"))```. There cleary is a foreign key, but `preg_match_all` in `getForeignKeyDetails` returns 0.

#### Expected behaviour

No exception is thrown.

**Comment by morozov** [comment]
This looks like a duplicate of https://github.com/doctrine/dbal/issues/5584 which should have been fixed by https://github.com/doctrine/dbal/pull/5597. After the fix, the DDL in the description matches the regular expression in `preg_match_all()`.

@troccoli could you check if your issue is reproducible on `^3.4@dev`?

**Comment by troccoli** [comment]
@morozov 

Thank you for a quick reply. Yes, I can confirm that the issue has been fixed in #5597 .

Do you know then 3.4.2 will be released?

**Comment by morozov** [comment]
Thanks for the confirmation, @troccoli. I plan to release it later today.
